### PR TITLE
adding my custom keypad to qmk repository

### PR DESCRIPTION
**Pull request by valdiieee**
i wanna add my firmware to qmk repository

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/makrosu/Makefile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ifndef MAKEFILE_INCLUDED
```
Comment (drashna) [inline keyboards/makrosu/Makefile:1]: This file should be removed, as it's deprecated.

File: keyboards/makrosu/MakrOSU.c [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "MakrOSU.h"
```
Comment (drashna) [inline keyboards/makrosu/MakrOSU.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#include "makrosu.h"
```

File: keyboards/makrosu/MakrOSU.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (drashna) [inline keyboards/makrosu/MakrOSU.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/makrosu/MakrOSU.h [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define KEYMAP( \
```
Comment (drashna) [inline keyboards/makrosu/MakrOSU.h:13]:
```suggestion
#define LAYOUT( \
```

File: keyboards/makrosu/config.h [hunk 5]
```diff
@@ -0,0 +1,83 @@
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/makrosu/config.h:2]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#pragma once
```

File: keyboards/makrosu/config.h [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xAB69
+#define PRODUCT_ID      0x8585
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Valdydesu_
+#define PRODUCT         MakrOSU
+#define DESCRIPTION     Macropad
```
Comment (drashna) [inline keyboards/makrosu/config.h:12]: This is deprecated. 
```suggestion
```

File: keyboards/makrosu/config.h [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 5
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 0
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 0
```
Comment (drashna) [inline keyboards/makrosu/config.h:39]:
```suggestion
```

File: keyboards/makrosu/config.h [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
```
Comment (drashna) [inline keyboards/makrosu/config.h:50]:
```suggestion
```

File: keyboards/makrosu/config.h [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/makrosu/config.h:53]: This setting is deprecated, as it's the default behavior.
```suggestion
```

File: keyboards/makrosu/config.h [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+// and avoid loading invalid data from the EEPROM
+#define EEPROM_VERSION 0x08
+#define EEPROM_VERSION_ADDR 34
+
+// Dynamic keymap starts after EEPROM version
+#define DYNAMIC_KEYMAP_EEPROM_ADDR 35
+// Dynamic macro starts after dynamic keymaps (35+(4*10*6*2)) = (35+480)
+#define DYNAMIC_KEYMAP_MACRO_EEPROM_ADDR 67    // **** CHANGE THIS BASED ON MATRIX_ROWS & MATRIX_COLS ****
+#define DYNAMIC_KEYMAP_MACRO_EEPROM_SIZE 957    // **** CHANGE THIS BASED ON 1024-DYNAMIC_KEYMAP_MACRO_EEPROM_ADDR ****
+#define DYNAMIC_KEYMAP_MACRO_COUNT 16
```
Comment (drashna) [inline keyboards/makrosu/config.h:73]: this is deprecated, and should be removed. 
```suggestion
```

File: keyboards/makrosu/config.h [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+#endif
```
Comment (drashna) [inline keyboards/makrosu/config.h:83]:
```suggestion
```

File: keyboards/makrosu/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,122 @@
+#include "MakrOSU.h"
```
Comment (drashna) [inline keyboards/makrosu/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/makrosu/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case OSU:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_OSU);
+	writePinHigh(IND_1);
+	writePinLow(IND_2);
+ 	writePinLow(IND_3);
```
Comment (drashna) [inline keyboards/makrosu/keymaps/default/keymap.c:56]: You could move the pin settings to:
```c
layer_state_t default_layer_state_set_user(layer_state_t state) {
    // indicators
    return state;
}
```

File: keyboards/makrosu/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,122 @@
+#include "MakrOSU.h"
```
Comment (drashna) [inline keyboards/makrosu/keymaps/via/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/makrosu/rules.mk [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+ENCODER_ENABLE = yes
+SRC += keyboards/wilba_tech/wt_main.c
+
+RAW_ENABLE = yes
+DYNAMIC_KEYMAP_ENABLE = yes
```
Comment (drashna) [inline keyboards/makrosu/rules.mk:24]:
```suggestion
BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
MOUSEKEY_ENABLE = no	# Mouse keys
EXTRAKEY_ENABLE = yes	# Audio control and System control
CONSOLE_ENABLE = no	# Console for debug
COMMAND_ENABLE = no    # Commands for debug and configuration
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = no
ENCODER_ENABLE = yes

VIA_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/makrosu/Makefile (hunk 1 above)
Comment (valdiieee) [reply keyboards/makrosu/Makefile]: ok done

**Comment by drashna** [comment]
Also, could you make sure that the folder and files are all lower case?

File: keyboards/makrosu/keymaps/default/keymap.c (hunk 13 above)
Comment (valdiieee) [reply keyboards/makrosu/keymaps/default/keymap.c]: done

**Review by valdiieee — COMMENTED** [review]
Its done

**Review by EmmStewie2709 — APPROVED** [review]

**Review by EmmStewie2709 — APPROVED** [review]

**Comment by valdiieee** [comment]
What should i do next?

File: keyboards/makrosu/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+            return state;
+            break;
+        case _3:
+            writePinLow(IND_1);
+            writePinLow(IND_2);
+	        writePinHigh(IND_3);
+            return state;
+            break;
+            
+    }
```
Comment (drashna) [inline keyboards/makrosu/keymaps/default/keymap.c:83]: This won't ever run the LED code. 
```suggestion
    state = update_tri_layer_state(state, _4, X_PAUSE, X_PAUSE);
    writePin(IND_1, layer_state_cmp(state, 1));
	writePin(IND_2, layer_state_cmp(state, 2));
 	writePin(IND_3, layer_state_cmp(state, 3)); 
 	return state;
```

File: keyboards/makrosu/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+            return state;
+            break;
+        case _3:
+            writePinLow(IND_1);
+            writePinLow(IND_2);
+	        writePinHigh(IND_3);
+            return state;
+            break;
+            
+    }
```
Comment (drashna) [inline keyboards/makrosu/keymaps/via/keymap.c:83]:
```suggestion
    state = update_tri_layer_state(state, _4, X_PAUSE, X_PAUSE);
    writePin(IND_1, layer_state_cmp(state, 1));
	writePin(IND_2, layer_state_cmp(state, 2));
 	writePin(IND_3, layer_state_cmp(state, 3)); 
 	return state;
```

File: keyboards/makrosu/rules.mk [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
+ENCODER_ENABLE = yes
+VIA_ENABLE = yes
```
Comment (drashna) [inline keyboards/makrosu/rules.mk:21]: This should only be enabled at the keymap level, not the keyboard level. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by valdiieee** [comment]
ok done, any other change?

**Review by drashna — APPROVED** [review]
LGTM

File: keyboards/makrosu/config.h [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
+
```
Comment (fauxpark) [inline keyboards/makrosu/config.h:62]:
```suggestion
```

File: keyboards/makrosu/rules.mk [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = no	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/makrosu/rules.mk:20]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/makrosu/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+  _1,
+  _2,
+  _3,
+  _4
+
+};
+
+
+enum planck_keycodes {
+  L1,
```
Comment (tzarc) [inline keyboards/makrosu/keymaps/default/keymap.c:29]:
```suggestion
  L1 = SAFE_RANGE,
```

File: keyboards/makrosu/keymaps/default/keymap.c [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+    setPinOutput(IND_3);
+    //set to off
+    writePinHigh(IND_1);
+    writePinHigh(IND_2);
+    writePinHigh(IND_3);
+}
+
+void matrix_scan_user(void) {
+}
+
```
Comment (tzarc) [inline keyboards/makrosu/keymaps/default/keymap.c:111]:
```suggestion
```

File: keyboards/makrosu/keymaps/via/keymap.c [hunk 23]
```diff
@@ -0,0 +1,136 @@
...
+  _1,
+  _2,
+  _3,
+  _4
+
+};
+
+
+enum planck_keycodes {
+  L1,
```
Comment (tzarc) [inline keyboards/makrosu/keymaps/via/keymap.c:29]:
```suggestion
  L1 = SAFE_RANGE,
```

File: keyboards/makrosu/keymaps/via/keymap.c [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+    setPinOutput(IND_3);
+    //set to off
+    writePinHigh(IND_1);
+    writePinHigh(IND_2);
+    writePinHigh(IND_3);
+}
+
+void matrix_scan_user(void) {
+}
+
```
Comment (tzarc) [inline keyboards/makrosu/keymaps/via/keymap.c:111]:
```suggestion
```

**Review by tzarc — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**